notes/src/View.js

```diff
@@ -3,61 +3,87 @@ import './index.css';
 import Note from "./Note";
 import Form from "./Form";
 import {connect} from "react-redux";
-import {getNotes} from "./Actions";
+import {getNotes,deleteNote} from "./Actions";
 import {Link} from "react-router-dom";
 
+
+const customStylesHidden =  {
+      top                   : '50%',
+      left                  : '50%',
+      right                 : 'auto',
+      bottom                : 'auto',
+      marginRight           : '-50%',
+      transform             : 'translate(-50%, -50%)',
+      color                 : "lightgray",
+      display               :  "none"
+  };
+
+  const customStyles = {
+      top                   : '50%',
+      left                  : '50%',
+      right                 : 'auto',
+      bottom                : 'auto',
+      marginRight           : '-50%',
+      transform             : 'translate(-50%, -50%)',
+      color                 : "lightgray",
+      display               :  "block"
+  };
+  
+  
+
 class View extends Component{
+    constructor(){
+        super();
+        this.state = {
+            display : customStylesHidden
+        }
+    }
 
     componentDidMount(){
+        console.log("View","componentDidMount")
         this.props.getNotes();
```

Comment:
Instead of getting all the notes and using the find method in your render function, try to use an axios get request to get the specific note. you'd have to add that to your actions! In your reducer, have initialState include `currentNote: undefined` and then an action that sets the note to the payload from that request. In componentDidMount do `this.props.getNote(this.props.match.params.id)`, and in mapStateToProps bring in currentNote. You can check for the existence of currentNote in your render. So, instead of `if(this.props.notes.length===0)` to return "Loading...", check for `if(!this.props.currentNote)`